astropy/conftest.py

```diff
@@ -23,7 +23,12 @@
     # This is a workaround for the OpenSSL deprecation warning that comes from
     # the `requests` module. It only appears when both asdf and sphinx are
     # installed. This can be removed once pyopenssl 1.7.20+ is released.
-    modules_to_ignore_on_import=['requests'])
+    modules_to_ignore_on_import=['requests'],
+    warnings_to_ignore_by_pyver={
```

Comment:
It is still unclear to me when to add it here and when to add it to

https://github.com/astropy/astropy/blob/4b7fcafa0b3b4e30d10ce9ffe60aece4a6ef3675/astropy/tests/helper.py#L137

Reply:
I'm on the side of we should set the very minimum of it in `astropy/astropy/tests/helper.py`, so downstream won't rely on it (and be surprised when we clean that file). I'm pretty sure there'll be a mpl version before 3.8 becomes widely used, so this should be a rather short term workaround anyway.